src/proxy/outbound.rs

```diff
@@ -397,23 +471,29 @@ impl OutboundConnection {
                     intended_destination_service: us.destination_service.clone(),
                     actual_destination,
                     upstream_sans,
+                    final_sans: vec![],
                 });
             }
             // Workload doesn't have a waypoint; send directly
         }
 
         // only change the port if we're sending HBONE
         let actual_destination = match us.workload.protocol {
-            Protocol::HBONE => SocketAddr::from((us.selected_workload_ip, self.hbone_port)),
+            Protocol::HBONE | Protocol::DOUBLEHBONE => {
+                SocketAddr::from((us.selected_workload_ip, self.hbone_port))
+            }
             Protocol::TCP => us.workload_socket_addr(),
         };
         let hbone_target_destination = match us.workload.protocol {
-            Protocol::HBONE => Some(us.workload_socket_addr()),
+            Protocol::HBONE | Protocol::DOUBLEHBONE => Some(us.workload_socket_addr()),
             Protocol::TCP => None,
         };
+        let (upstream_sans, final_sans) = match us.workload.protocol {
```

Comment:
My understanding from talking to @keithmattix is that `Workload.subjectAlternativeNames` will be repurposed to contain the identities of remote pods/waypoints, so I should change the logic of the other protocols to only use `us.workload.identity` instead of `us.workload_and_services_san`.

Reply:
Yes, I think this is correct; only the double hbone codepath needs to be added/changed because there are two sans being considered: the e/w gateway SAN and the SANs of the backends. So what you have looks right to me